source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -1,28 +1,76 @@
 #include "extensions/filters/common/local_ratelimit/local_ratelimit_impl.h"
 
+#include "common/protobuf/utility.h"
+
 namespace Envoy {
 namespace Extensions {
 namespace Filters {
 namespace Common {
 namespace LocalRateLimit {
 
-LocalRateLimiterImpl::LocalRateLimiterImpl(const std::chrono::milliseconds fill_interval,
-                                           const uint32_t max_tokens,
-                                           const uint32_t tokens_per_fill,
-                                           Event::Dispatcher& dispatcher)
+LocalRateLimiterImpl::LocalRateLimiterImpl(
+    const std::chrono::milliseconds fill_interval, const uint32_t max_tokens,
+    const uint32_t tokens_per_fill, Event::Dispatcher& dispatcher,
+    const Envoy::Protobuf::RepeatedPtrField<
+        envoy::extensions::common::ratelimit::v3::LocalRateLimitDescriptor>& descriptors,
+    bool per_route)
     : fill_interval_(fill_interval), max_tokens_(max_tokens), tokens_per_fill_(tokens_per_fill),
       fill_timer_(fill_interval_ > std::chrono::milliseconds(0)
                       ? dispatcher.createTimer([this] { onFillTimer(); })
-                      : nullptr) {
+                      : nullptr),
+      time_source_(dispatcher.timeSource()) {
   if (fill_timer_ && fill_interval_ < std::chrono::milliseconds(50)) {
     throw EnvoyException("local rate limit token bucket fill timer must be >= 50ms");
   }
 
-  tokens_ = max_tokens;
+  tokens_.tokens = max_tokens;
 
   if (fill_timer_) {
     fill_timer_->enableTimer(fill_interval_);
   }
+
+  if (per_route) {
```

Comment:
Why is this only done for the per_route case? I'm not sure why per_route is passed to this in general. I don't think this object should care where it's used? It should either have descriptor settings or not?

Reply:
Because we will get descriptors in routes config in filter.. hence this was added..

Reply:
Sorry I don't understand. The filter can either have a global config or a per route config with a limiter defined for either, right? Why do we need to understand per-route here?

Reply:
And descriptors will only work for per route config as at runtime we get them through rate limit actions in route components proto

Reply:
That doesn't make sense. I don't think you need per_route_config for this to work. You can have a global config for the filter which still references a route table. I think this should be removed here.

Reply:
Removed